code/test.py

```diff
@@ -252,50 +199,44 @@ def do_tests():
-                print('float32 expected', expected_times_32, file=sys.stderr)
-                print('float32 % expected/get', (
-                    expected_times_32 / float32_times), file=sys.stderr)
             if do_gpu:
                 print('gpu times', gpu_times, file=sys.stderr)
-                print('gpu expected', expected_times_gpu, file=sys.stderr)
-                print('gpu % expected/get', (
-                    expected_times_gpu / gpu_times), file=sys.stderr)
 
             print()
             if do_float64 and do_float32:
                 print('float64/float32', (
                     float64_times / float32_times), file=sys.stderr)
-                print('expected float64/float32', (
-                    expected_times_64 / float32_times), file=sys.stderr)
             if do_float64 and do_gpu:
                 print('float64/gpu', float64_times / gpu_times, file=sys.stderr)
-                print('expected float64/gpu', (
-                    expected_times_64 / gpu_times), file=sys.stderr)
             if do_float32 and do_gpu:
                 print('float32/gpu', float32_times / gpu_times, file=sys.stderr)
-                print('expected float32/gpu', (
-                    expected_times_32 / gpu_times), file=sys.stderr)
-
-    def compare(x, y):
-        ratio = x / y
-        # If there is more then 5% difference between the expected
-        # time and the real time, we consider this an error.
-        return sum((ratio < 0.95) + (ratio > 1.05))
-
-    print(file=sys.stderr)
-    if do_float64:
-        err = compare(expected_times_64, float64_times)
-        print('speed_failure_float64=' + str(err), file=sys.stderr)
-    if do_float32:
-        err = compare(expected_times_32, float32_times)
-        print('speed_failure_float32=' + str(err), file=sys.stderr)
-    if do_gpu:
-        err = compare(expected_times_gpu, gpu_times)
-        print('speed_failure_gpu=' + str(err), file=sys.stderr)
-
-        assert not numpy.isnan(gpu_times).any()
+        
+    # Write JUnit xml for speed test performance report
+
+    speed_file = 'speedtests_time.xml'
+
+    # Define speed test file write method
+    def write_junit(filename, algos, times, label):
+        with open(filename, 'a') as f:
+            for algo, time in zip(algos, times):
+                f.write('   <testcase classname="{label}" name="{algo}" time="{time}">'
+                        .format(label=label, algo=algo, time=time))
+                f.write('   </testcase>\n')
+
+    test_total = numpy.size(float64_times) \
+                 + numpy.size(float32_times) \
+                 + numpy.size(gpu_times)
+
+    with open(speed_file, 'w') as f:
+        f.write('<?xml version="1.0" encoding="UTF-8"?>\n')
+        f.write('<testsuite name="theano_speedtests" tests="{ntests}">\n'
+                .format(ntests=numpy.size(test_total)))
+
+    write_junit(speed_file, algo_executed, float64_times, label='float64')
```

Comment:
Done in new commits, also corrected some errors that would pop up if some tests flags were set to false, it would still to access the undeclared results variables.